Shopify/ruby-lsp
# Refactor code actions to use context instead of our own cache

### Motivation

Closes #487 

Instead of caching diagnostics ourselves, we should leverage the specification and use what's already available. Every code action request receives a `context` object, which contains the current diagnostics for the file, so there's no point in reading from our cache.

**Note**: the reason this is coming during this week is because it helps my hackdays idea 😂.

### Implementation

- Removed a ton of unused code, especially related to syntax errror diagnostics which we no longer produce
- Started returning the code actions straight from the diagnostics request inside the `data` attribute which is always preserved between a diagnostic -> code action request
- Simplified code actions to use the `context` parameter rather than trying to read from our own cache

### Automated Tests

Had to change a bunch of tests. Some didn't make sense anymore, others needed to have expectations updated since the response changed for diagnostics and code actions.

### Manual Tests

Verify code actions still work. For example
1. Open the LSP on this branch
2. Make a violation (e.g.: wrong indentation)
3. Hover over the violation
4. Use the quickfix
5. Verify that the violation has been fixed
